src/jestUtils.ts

```diff
@@ -0,0 +1,388 @@
+  }
+  throw Error(
+    'Unable to resolve gesture handler tag, are you sure that you added {...ghTagEventMacro()} to your component?'
+  );
+};
+
+export const fireTapGestureHandler = (
+  component: any,
+  eventData?: FireGestureHandlerConfig<TapConfig>
+) => {
+  const baseEventData: Required<TapConfig> = {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+  };
+  runEventsSequence(
+    component,
+    GestureHandlerType.TAP,
+    baseEventData,
+    eventData
+  );
+};
+
+export const firePanGestureHandler = (
+  component: any,
+  eventData?: FireGestureHandlerConfig<PanConfig>
+) => {
+  const baseEventData: Required<PanConfig> = {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    translationX: 0,
+    translationY: 0,
+    velocityX: 0,
+    velocityY: 0,
+  };
+  runEventsSequence(
+    component,
+    GestureHandlerType.PAN,
+    baseEventData,
+    eventData
+  );
+};
+
+export const fireLongPressGestureHandler = (
+  component: any,
+  eventData?: FireGestureHandlerConfig<LongPressConfig>
+) => {
+  const baseEventData: Required<LongPressConfig> = {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+    duration: 0,
+  };
+  runEventsSequence(
+    component,
+    GestureHandlerType.LONG_PRESS,
+    baseEventData,
+    eventData
+  );
+};
+
+export const fireRotationGestureHandler = (
+  component: any,
+  eventData?: FireGestureHandlerConfig<RotationConfig>
+) => {
+  const baseEventData: Required<RotationConfig> = {
+    rotation: 0,
+    velocity: 0,
+    anchorX: 0,
+    anchorY: 0,
+  };
+  runEventsSequence(
+    component,
+    GestureHandlerType.ROTATION,
+    baseEventData,
+    eventData
+  );
+};
+
+export const fireFlingGestureHandler = (
+  component: any,
+  eventData?: FireGestureHandlerConfig<RotationConfig>
+) => {
+  const baseEventData: Required<TapConfig> = {
+    x: 0,
+    y: 0,
+    absoluteX: 0,
+    absoluteY: 0,
+  };
+  runEventsSequence(
+    component,
+    GestureHandlerType.FLING,
+    baseEventData,
+    eventData
+  );
+};
+
+export const firePinchGestureHandler = (
+  component: any,
+  eventData?: FireGestureHandlerConfig<RotationConfig>
+) => {
+  const baseEventData: Required<PinchConfig> = {
+    scale: 0,
+    velocity: 0,
+    focalX: 0,
+    focalY: 0,
+  };
+  runEventsSequence(
+    component,
+    GestureHandlerType.PINCH,
+    baseEventData,
+    eventData
+  );
+};
+
+export const decorateChildrenWithTag = (
+  component: any,
+  handlerProperties: HandlerProperties
+) => {
+  if (!component) return;
+  if (component.props.ghTagContainer !== undefined) {
+    component.props.ghTagContainer.push(handlerProperties);
+  }
+  if (Array.isArray(component.props.children)) {
+    for (const child of component.props.children) {
+      decorateChildrenWithTag(child, handlerProperties);
+    }
+  } else if (typeof component.props.children === 'object') {
+    decorateChildrenWithTag(component.props.children, handlerProperties);
+  }
```

Comment:
I think this may fail if you were to use your own component as a child of `GestureDetector`, then its inner structure wouldn't be decorated. In case of gesture handlers it shouldn't matter that much since they required a native view as a direct child.